metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -255,6 +255,25 @@ final class ReferenceProvider(
     }
   }
 
+  private[metals] def pathsMightContainSymbol(
+      source: AbsolutePath,
+      isSymbol: Set[String],
+  ): Set[AbsolutePath] = {
+    buildTargets.inverseSources(source) match {
```

Comment:
Can we use `buildTargets.inverseSourcesBsp`? It's has been recently added because in some cases plain `inverseSources` might not be enough. It's more awkward  to handle since it brings future to the scope but I see that it's already present in `CallHierarchyProvider` so issue shouldn't be too big. 
`inverseSourcesBsp` was added in https://github.com/scalameta/metals/pull/3752, but I can't see followup issue with call migration, is there a chance that there isn't one? cc: @tanishiking